pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -159,6 +161,19 @@ func migratableDomXML(dom cli.VirDomain, vmi *v1.VirtualMachineInstance) (string
 				inMetaKVMigration = true
 			}
 			depth++
+
+			// If the VMI requires dedicated CPUs, we need to patch the domain with
+			// the new CPU set in the target node prior to migration
+			if vmi.IsCPUDedicated() {
```

Comment:
I think we should consider all cases we handle in [1]
Perhaps the best is to separate this block into a function and feed it with a CPUSET. In this case, it would be `TargetNodeCPUSet`
This way we will be able to generate the same results for both source and the destination.
[1] https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/converter/converter.go#L1628

Reply:
+1, I'm responsible for this ugly parser, and it would be nice to avoid adding even more complexity to it as much as possible...

Also, adding at least a depth check to the `if` above would ensure we're somewhat in the right place, and avoid running the loop below for every StartElement.
Even better would be to ensure we're exactly in the right place by using booleans like the `inMeta...` above.

If this is going to be used more and more, maybe we should tidy it up and generalize it, by replacing all the bools and the depth by a single string slice. Let me know if that's something of interest, I'm willing to spend some time on this to make it nicer.

Reply:
Thanks for the review Jed! I am now investigating a new approach to this, I am trying to see if I can hook to the migration proxy on the target node and patch the domain spec there by reusing the ConverterContext functionality that already exists, this way not deviating from Roman's new implementation of CPU pinning, WDYT?
@rmohr @jean-edouard @vladikr

Reply:
The destination domain xml is being transmitted from the source, it will be too late when it will get to the target, as the migration will already be in progress. If we won't for the destination xml, the source will simply set the exiting source xml as the destination. I'm not aware of a way to patch the destination xml on the destination.

From my point of view, we need to separate https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/converter/converter.go#L1586-L1624 into its own function that will be able to recive node topology, guest topology and a CPUSET as parameters so we could use this to form both the cpu layout for the source and the target. 
What do you think?

Reply:
> The destination domain xml is being transmitted from the source, it will be too late when it will get to the target, as the migration will already be in progress. If we won't for the destination xml, the source will simply set the exiting source xml as the destination. I'm not aware of a way to patch the destination xml on the destination.

According to  https://libvirt.org/hooks.html#qemu libvirt has at least a hook. Not sure if there is a programmatic way outside of that hook.